lib/bin/lwm2m_carrier/os/lwm2m_os.c

```diff
@@ -226,7 +225,7 @@ void lwm2m_os_timer_cancel(void *timer)
 		return;
 	}
 
-	k_delayed_work_cancel(&work->work_item);
+	k_work_cancel_delayable(&work->work_item);
```

Comment:
This may fail, and if it does, we'll keep scheduling and rescheduling the timer anyway. According to [best practices](https://docs.zephyrproject.org/latest/reference/kernel/threads/workqueue.html#check-return-values), these `cancel` calls should have a short comment explaining what happens if they fail.

Reply:
Alright. Note that this has undefined behavior if called from a work handler. I can't tell whether this can happen here, it might be worth looking into.

Reply:
Thank you, we'll handle this in the binary.